src/index.ts

```diff
@@ -39,22 +39,69 @@ function logError(msg) {
   spinner.fail();
 }
 
-function transformWebModuleFilename(depName: string): string {
-  return depName.replace('/', '--');
+class ErrorWithHint extends Error {
+  constructor(message: string, public readonly hint: string) {
+    super(message);
+  }
+}
+
+/**
+ * Resolve a "webDependencies" input value to the correct absolute file location.
+ * Supports both npm package names, and file paths relative to the node_modules directory.
+ * Follows logic similar to Node's resolution logic, but using a package.json's ESM "module"
+ * field instead of the CJS "main" field.
+ */
+function resolveWebDependency(dep: string): string {
+  const lazyDependencyPath = path.join(cwd, 'node_modules', dep);
+  let dependencyStats: fs.Stats;
+  try {
+    dependencyStats = fs.statSync(lazyDependencyPath);
+  } catch (err) {
+    throw new Error(`"${dep}" not found in your node_modules directory. Did you run npm install?`);
+  }
+  if (dependencyStats.isFile()) {
+    return lazyDependencyPath;
+  }
+  if (dependencyStats.isDirectory()) {
+    const dependencyManifestLoc = path.join(cwd, 'node_modules', dep, 'package.json');
+    const manifest = require(dependencyManifestLoc);
+    if (!manifest.module) {
+      throw new ErrorWithHint(
+        `dependency "${dep}" has no ES "module" entrypoint.`,
+        chalk.italic(
+          `Tip: Find modern, web-ready packages at ${chalk.underline(
+            'https://pikapkg.com/packages',
+          )}`,
+        ),
+      );
+    }
+    return path.join(cwd, 'node_modules', dep, manifest.module);
```

Comment:
you could re-use your existing variable:
`path.join(lazyDependencyPath, manifest.module)`